src/main.rs

```diff
@@ -21,39 +21,50 @@ pub mod pubsub;
 pub mod storage;
 
 use std::fs;
+use std::fs::{OpenOptions};
+use std::path::PathBuf;
 use std::thread;
 use std::sync::Arc;
-
-use std::sync::Mutex;
+use std::sync::mpsc;
+use memmap::{Mmap, Protection};
 
 lazy_static! {
-
  static ref ENVIRONMENT: lmdb::Environment = {
     let _ = config::set_default("storage.path", "pumpkin.db");
     let storage_path = config::get_str("storage.path").unwrap().into_owned();
     fs::create_dir_all(storage_path.as_str()).expect("can't create directory");
     let map_size = config::get_int("storage.mapsize");
     storage::create_environment(storage_path, map_size)
  };
+}
 
- static ref DATABASE: Arc<storage::Storage<'static>> = {
-    Arc::new(storage::Storage::new(&ENVIRONMENT))
- };
-
- static ref PUBLISHER: Mutex<pubsub::PublisherAccessor<Vec<u8>>> = {
-     let mut publisher = pubsub::Publisher::new();
-     let publisher_accessor = publisher.accessor();
-     let _ = thread::spawn(move || publisher.run());
-     Mutex::new(publisher_accessor)
- };
-
+/// Accepts storage path, filename and length and prepares the file. It is important that the length
+/// is the total length of the memory mapped file, otherwise the application _will segfault_ when
+/// trying to read those sections later. There is no way to handle that.
+fn prepare_mmap(storage_path: &str, filename: &str, length: u64) -> Mmap {
+    let mut scratchpad_pathbuf = PathBuf::from(storage_path);
+    scratchpad_pathbuf.push(filename);
+    scratchpad_pathbuf.set_extension("dat");
+    let scratchpad_path = scratchpad_pathbuf.as_path();
+    let scratchpad_file = OpenOptions::new().create_new(false).write(true)
```

Comment:
Actually, `true` isn't right either. It would work for the new one, but then it is resulting `File exists`. Therefore, it should be `create(true)`, I think.